ServerGUI.py

```diff
@@ -92,13 +116,54 @@ def nextLoc(self):
         d = math.sqrt((self.curr_tile.x - next_tile.x)**2 + (self.curr_tile.y - next_tile.y)**2)
         return d <= reached_tile_bound
 
+    def updateDesiredHeading(self, next_tile):
+        """
+        calculates the degrees between the current tile and the next tile and updates desired_heading. Estimates the
+        degrees to the nearing int.
+        """
+        x_change = next_tile.x - self.curr_tile.x
+        y_change = next_tile.y - self.curr_tile.y
+        if y_change == 0:
+            arctan = 90 if x_change < 0 else -90
+        else:
+            arctan = math.atan(x_change/y_change) * (180 / math.pi)
+        if x_change >= 0 and y_change > 0:
+            self.desired_heading = (360-arctan) % 360
+        elif x_change < 0 and y_change > 0:
+            self.desired_heading = -arctan
+        else:
+            self.desired_heading = 180 - arctan
+        self.desired_heading = round(self.desired_heading)
+
+    def computeMotorSpeed(self):
+        """
+        Currently assuming:
+            if desired angle > current angle, turn right
+            if desired angle < current angle, turn left
+            Threshold of 3 degrees, will only try to rotate if the rotation
+            is more than 3 degrees.
+        """
+        if (self.curr_tile.x, self.curr_tile.y) == self.endPoint and (self.desired_heading - self.heading <= 3) and (self.desired_heading - self.heading >= -3):
+            return ()
+        if self.desired_heading - self.heading > 3:
+            return rotation_right
+        elif self.desired_heading - self.heading < -3:
+            return rotation_left
+        else:
+            return motor_speed
+
+
     def main_loop(self):
         """
         """
         #  TODO 1: update location based on indoor GPS
         self.update_loc()
         self.drawC1C0()
-        self.server.send_update((self.curr_tile.row, self.curr_tile.col))
+        if self.run_mock:
+            self.server.send_update((self.curr_tile.row, self.curr_tile.col))
+        else:
+            data = self.computeMotorSpeed()
```

Comment:
never name a variable data. never. 
I like placing self.computeMotorSpeed() directly in the update function, but some programers like breaking up the code with variables